src/components/layout.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { CountriesContext } from '../contexts/countries';
+import { CountriesContextType, TypeCountry } from '../@types/country';
+import { useSorting } from '../hooks';
+import Button from './button';
+import Select from './select';
+import Countries from './countries';
+import IsLoading from './is-loading';
+import Error from './error';
+
+const Layout:React.FC = () =>{
+    const { countries, error, isLoading, getData } = React.useContext(CountriesContext) as CountriesContextType;
+    const [selectedContinent, setSelectedContinent] = React.useState<string | null>(null);
+    const [mobile, setMobile] = React.useState<boolean>(false);
+    
+    const query = window.matchMedia("(max-width:440px)");
+
+    const continents:string[] = Array.from(
+        new Set<string>(countries.map((country: TypeCountry) => country.continent))
+    );
```

Comment:
again here, split into named functions so I know what's happening. also no need to type cast (almost anything, ts will know!)